src/libraries/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/ObjectManager.cs

```diff
@@ -379,6 +384,19 @@ private bool DoValueTypeFixup(FieldInfo? memberToFix, ObjectHolder holder, objec
             return true;
         }
 
+        [DynamicDependency("value", typeof(Nullable<>))]
```

Comment:
Are these refactoring buying us anything? Wouldn't it be better to just slap `RequiresUnreferencedCode` on `DoValueTypeFixup`? I DoValueTypeFixup is called from FixupSpecialObject which already has this annotation anyway. Is there a public codepath that becomes trimming safe thanks to this?

Reply:
Btw, this pattern is `AddRef` to #45771 - it would be trimming safe without annotations if we had that API.

Reply:
> Is there a public codepath that becomes trimming safe thanks to this?

My initial thinking was that eventually there was a public code path that this allowed for, but looking again, if I add about 6 more `RequiresUnreferencedCode` attributes to private/internal methods this could be marked with `RequiresUnreferencedCode` with no changes to the public API.

However, I'm a bit torn because what this code is doing is technically not `RequiresUnreferencedCode`. It just needs to get a field that is very likely not ever trimmed. So I am kind of leaning towards leaving this.  Thoughts?

Reply:
> Thoughts?

I don't have an opinion. This code is likely going to be deleted in a couple years from now so probably "do whatever is cheapest".

> It just needs to get a field that is very likely not ever trimmed

True, Nullable is very special and the runtime is aware of its layout. But I would like us to be mindful of the distinction between "the field is going to be kept" and "the field name is going to be kept". Illink could potentially do an optimization to either completely remove, or shorten names of fields in the future if they're known not to be reflected on: https://github.com/mono/linker/issues/1282. I assume the `DynamicDependency` would ensure the name is not removed in this case.

Reply:
> Illink could potentially do an optimization to either completely remove, or shorten names of fields in the future if they're known not to be reflected on: mono/linker#1282. 

I think that decides it for me - I will keep this refactoring to ensure that the `value` field of `Nullable<>` is not mucked with.